src/bdecode.cpp

```diff
@@ -289,6 +289,93 @@ namespace {
 		m_buffer = buf;
 	}
 
+	bool bdecode_node::has_soft_error(span<char> error) const
+	{
+		bdecode_token const* tokens = m_root_tokens;
+		int token = m_token_idx;
+
+		// we don't know what the original depth_limit was
+		// so this has to go on the heap
+		std::vector<int> stack;
+		// make the initial allocation the default depth_limit
+		stack.reserve(100);
+
+		do
+		{
+			switch (tokens[token].type)
```

Comment:
I think this function should have an early exit for the case of the ``bdecode_node`` object being default constructed (or in the state after ``clear()`` is called).

When the node is not the root, ``m_tokens`` will be empty, so it's not is simple as just checking that. Perhaps a check that ``m_buffer`` is set? or perhaps ``m_token_idx`` >= 0.